File: LeetCode Problems/101 to 200/151. Reverse Words in a String/151. Reverse Words in a String.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        String words[] = s.split(" ");
        s = "";
        for (int i = words.length - 1; i >= 0; i--) {
            if (words[i].trim().length() == 0) continue;
            s += words[i] + (i == 0 ? "" : " ");
        }
        return s;
    }
}

File: LeetCode Problems/101 to 200/151. Reverse Words in a String/151. Reverse Words in a String .cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words;
        string str = "";
        for (int i = 0; i < s.size(); i++) {
            if (s[i] != ' ') {
                str += s[i];
            } else if (s[i] == ' ' && str.size() > 0) {
                words.push_back(str);
                str = "";
            }
        }
        if (str.size() > 0) {
            words.push_back(str);
        }
        reverse(words.begin(), words.end());
        str = words[0];
        for (int i = 1; i < words.size(); i++) {
            str += " ";
            str += words[i];
        }
        return str;
    }
};
